scripts/get_objf_and_derivs_split.py

```diff
@@ -351,6 +354,57 @@ def MergeAndComputeObjfForSplit(split_index):
     MergeAllOrders(split_index)
     ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None)
 
+def CleanUpIfNeeded():
+    # For cleaning up intermediate files that won't be used, note if option need-model = true
+    # be used, clean up will keep file 'float.all'
+    if args.clean_up == 'true':
+        try:
+            if not os.path.isdir(args.work_dir):
+                ExitProgram("error finding working directory {0}".format(args.work_dir))
+            # If need_model option set to be false, this script will keep "float.all" in
+            # each split dir for future use
+            need_split_float = True if args.need_model == 'false' else False
+            for split_index in range(1, args.num_splits + 1):
```

Comment:
Maybe the "a" is just a typo.
I'm not sure the best way to if dir is empty in python, I'm new to python-
try google.
Dan


On Tue, Jul 12, 2016 at 11:44 AM, chris920820 <notifications@github.com>
wrote:

> In scripts/get_objf_and_derivs_split.py
> <https://github.com/danpovey/pocolm/pull/30#discussion_r70497747>:
>
> > @@ -351,6 +354,57 @@ def MergeAndComputeObjfForSplit(split_index):
> >      MergeAllOrders(split_index)
> >      ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None)
> >
> > +def CleanUpIfNeeded():
> > +    # For cleaning up intermediate files that won't be used, note if option need-model = true
> > +    # be used, clean up will keep file 'float.all'
> > +    if args.clean_up == 'true':
> > +        try:
> > +            if not os.path.isdir(args.work_dir):
> > +                ExitProgram("error finding working directory {0}".format(args.work_dir))
> > +            # If need_model option set to be false, this script will keep "float.all" in
> > +            # each split dir for future use
> > +            need_split_float = True if args.need_model == 'false' else False
> > +            for split_index in range(1, args.num_splits + 1):
>
> Indeed, actually I have thought this for a while, but hesitating of adding
> a “unnecessary” options. Question i) what is the best way to check if a dir
> is empty in python I do something like if os.listdir(some_dir) == []: but
> not sure if there is some potential problem (the first one I can think of
> is that is this dir contains some file start with ‘.’, second is if this
> dir contains many files, this can be very time consuming, maybe we want it
> return “false” immediately after it finding the first file. ii) When
> running get_counts.py, it print the following info in stderr get_counts.py:
> dumping counts a a “a” is printed by purpose or something symbolized “a
> literation”? thanks for your advice again
> … <#m_3706537280204711803_>
> On Jul 11, 2016, at 6:57 PM, Daniel Povey ***@***.***> wrote: In
> scripts/get_objf_and_derivs_split.py: > @@ -351,6 +354,57 @@ def
> MergeAndComputeObjfForSplit(split_index): > MergeAllOrders(split_index) >
> ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None) > > +def
> CleanUpIfNeeded(): > + # For cleaning up intermediate files that won't be
> used, note if option need-model = true > + # be used, clean up will keep
> file 'float.all' > + if args.clean_up == 'true': > + try: > + if not
> os.path.isdir(args.work_dir): > + ExitProgram("error finding working
> directory {0}".format(args.work_dir)) > + # If need_model option set to be
> false, this script will keep "float.all" in > + # each split dir for future
> use > + need_split_float = True if args.need_model == 'false' else False >
> + for split_index in range(1, args.num_splits + 1): this logic doesn't seem
> right to me. Typically when this script is called, it's just get the
> derivatives w.r.t. some scalar values, so the split float.all files won't
> be used. You should probably add a separate boolean command-line argument
> --need-split-model which can be set from make_lm_dir.py in the case where
> it needs the split model. — You are receiving this because you authored the
> thread. Reply to this email directly, view it on GitHub, or mute the
> thread.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/30/files/46f9c831414495f7bcc26fec087abf16f90c403a#r70497747>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADJVu1iZ21uk08BRdfPdotClu26NoWBiks5qU-CogaJpZM4JI_LN>
> .
>